BoxContentSDK/BoxContentSDK/Models/BOXRepresentation.m

```diff
@@ -41,27 +46,26 @@ - (instancetype)initWithJSON:(NSDictionary *)JSONResponse
                                                                     nullAllowed:NO];
         self.details = detailsJSON;
         
-        NSDictionary *linksJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyLinks
+        NSDictionary *contentJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyContent
                                                                  inDictionary:JSONResponse
                                                               hasExpectedType:[NSDictionary class]
                                                                   nullAllowed:NO];
         
-        NSDictionary *contentJSON = linksJSON[BOXAPIObjectKeyContent];
-        
-        NSString *contentURLString = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyURL
+        NSString *contentURLString = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyURLTemplate
                                                              inDictionary:contentJSON
                                                           hasExpectedType:[NSString class]
                                                               nullAllowed:NO];
         if (contentURLString.length > 0) {
+            if ([self.type isEqualToString:BOXRepresentationTypeHLS]) {
```

Comment:
URLWithString will set contentURL to nil for this failure case. 

But this brings up a good point, we are not handling mp4. Can I append {+asset_path} as 'preview.mp4' for mp4 reps?
Ex:
"url_template": "https:\/\/dl3.boxcloud.com\/api\/2.0\/internal_files\/125041766103\/versions\/133811379927\/representations\/mp4\/content\/{+asset_path}"

With contentURL as nil means we are currently we are failing over to our old api call.